gammapy/datasets/simulate.py

```diff
@@ -169,6 +169,12 @@ def _sample_coord_time_energy(self, dataset, model):
 
         npred = self._evaluate_timevar_source(dataset, model=model)
         data = npred.data[np.isfinite(npred.data)]
+
+        if np.sum(data) > 1e6:
+            raise TypeError(
+                f"The number of predicted events for the model {model.name} is too large. No event sampling will be performed for this model!"
```

Comment:
Another approach is to try the poisson random and catch the exception to raise the one you want with its correct message.